datamonster_api/lib/datamonster.py

```diff
@@ -156,6 +156,19 @@ def get_datasources(self, query=None, company=None):
         datasources = self._get_paginated_results(url)
         return six.moves.map(self._datasource_result_to_object, datasources)
 
+    def get_datasource_by_name(self, name):
+        """Given a name, try to find a datasource"""
+        result = list(self.get_datasources(query=name))
```

Comment:
What if we have a data source called "1010 Data Credit Sales Index" and another called "1010 Data Credit Sales Index YoY"? If I send the query by name "1010 Data Credit Sales Index", won't there be an error because two matched even though I provided the exact name?

Maybe you want to filter result checking for an exact match to name before checking the length? Extra question, should the name search be case sensitive in that comparison?

Reply:
we're using ilike on the filtering side, so makes sence for this to be case insensitive also
```
        if q:
            datasources = datasources.filter(DataSource.name.ilike('%{}%'.format(q)))
```

updated